src/utilities/__tests__/extendSchema-test.js

```diff
@@ -1286,6 +1302,31 @@ describe('extendSchema', () => {
       );
     });
 
+    it('adds schema definition missing in the original schema', () => {
+      let schema = new GraphQLSchema({
+        directives: [
+          new GraphQLDirective({
+            name: 'onSchema',
+            locations: ['SCHEMA'],
+          }),
+        ],
+      });
+      expect(schema.getQueryType()).to.equal(undefined);
+
+      const ast = parse(`
+        schema @onSchema {
+          query: Foo
+        }
+
+        type Foo {
+          bar: String
+        }
+      `);
+      schema = extendSchema(schema, ast);
+      const queryType = schema.getQueryType();
+      expect(queryType).to.have.property('name', 'Foo');
+    });
```

Comment:
@mjmahone This is proposed solution for a situation where library/tools wants to add its directives to user-provided SDL.
Alternatively, you can just prefix user-provided SDL but it will screw line offsets in errors.
Another alternative is to use `concatAST` but there no method to convert `GraphQLDirective` to AST.

This is a pretty big change but in addition to injecting directives it also allows to implement `buildASTSchema` as follows:
```js
function buildASTSchema(ast, options) {
  return extendSchema(new GraphQLSchema({}), ast, options);
}
```

To make review easier tomorrow I will extract this change into separate PR.

Reply:
That looks pretty cool! I think we might need to provide a method to convert `GraphQLDirective` to AST.